airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -37,12 +60,66 @@ class SourceZendeskSupportStream(HttpStream, ABC):
 
     transformer = TypeTransformer(TransformConfig.DefaultSchemaNormalization)
 
-    def __init__(self, subdomain: str, **kwargs):
+    def __init__(self, subdomain: str, authenticator: Union[AuthBase, HttpAuthenticator] = None, **kwargs):
         super().__init__(**kwargs)
+        self._session = SourceZendeskSupportFuturesSession()
+        self._session.auth = authenticator
 
         # add the custom value for generation of a zendesk domain
         self._subdomain = subdomain
 
+    def _send(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+        response: requests.Response = self._session.send(request, **request_kwargs)
+        return response
+
+    def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+        return self._send(request, request_kwargs)
+
+    def read_records(
```

Comment:
this is repeating the whole CDK method, can you add some context on why? ideally this should not be the case